### PR TITLE
Move config parsing to use context

### PR DESCRIPTION
**Pull request by jackzampolin**
This PR will move the config parsing to stuffing the chains into the `context.Context` where they will be consumed in the cli functions

### CODE REVIEW

File: client/context.go [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+	"context"
+	"fmt"
+)
+
+type contextKey string
+
+func newContextKey(chainid string) contextKey {
+	return contextKey(fmt.Sprintf("chain-client-context/%s", chainid))
+}
+
```
Comment (joeabbey) [inline client/context.go:13]: Comments on exported method

```suggestion
//SetChainClientOnContext sets ChainClient on Context
```

**Review by joeabbey — APPROVED** [review]
LGTM

File: cmd/modules.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+	"github.com/cosmos/cosmos-sdk/x/mint"
+	"github.com/cosmos/cosmos-sdk/x/params"
+	paramsclient "github.com/cosmos/cosmos-sdk/x/params/client"
+	"github.com/cosmos/cosmos-sdk/x/slashing"
+	"github.com/cosmos/cosmos-sdk/x/staking"
+	"github.com/cosmos/cosmos-sdk/x/upgrade"
+	upgradeclient "github.com/cosmos/cosmos-sdk/x/upgrade/client"
+)
+
+var ModuleBasics = []module.AppModuleBasic{
```
Comment (marbar3778) [inline cmd/modules.go:23]: is this for encodings or decoding txs?

File: cmd/config.go [hunk 3]
```diff
@@ -111,22 +120,42 @@ func defaultConfig(keyHome string, debug bool) []byte {
...
+				GRPCAddr:       "https://gprc.cosmoshub-4.technofractal.com:443",
+				AccountPrefix:  "cosmos",
+				KeyringBackend: "test",
+				GasAdjustment:  1.2,
+				GasPrices:      "0.01uatom",
+				KeyDirectory:   keyHome,
+				Debug:          debug,
+				Timeout:        "20s",
+				OutputFormat:   "json",
+				BroadcastMode:  "block",
```
Comment (marbar3778) [inline cmd/config.go:138]: should avoid using block, this was meant for testing that ended up bleeding into prod.

File: cmd/modules.go (hunk 2 above)
Comment (jackzampolin) [reply cmd/modules.go:23]: Both

File: cmd/config.go (hunk 3 above)
Comment (jackzampolin) [reply cmd/config.go:138]: oh interesting

Comment (jackzampolin) [reply cmd/config.go:138]: I guess I should add a mempool query that sees if we have a TX in the mempool before trying to submit another one

**Comment by jackzampolin** [comment]
closing this in favor of other work. Come back to this once cobra merges https://github.com/spf13/cobra/pull/1551

**Pull request closed without merging**